pkg/agent/monitortool/latency_store.go

```diff
@@ -0,0 +1,218 @@
+// Copyright 2024 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package monitortool
+
+import (
+	"errors"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/containernetworking/plugins/pkg/ip"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+
+	coreinformers "k8s.io/client-go/informers/core/v1"
+)
+
+type LatencyStore struct {
+	// Maybe we need to use small lock for the map
+	mutex sync.RWMutex
+
+	// The map of node name to node info, it will changed to gw0 watcher
+	nodeInformer coreinformers.NodeInformer
+	// The map of node ip to connection
+	connectionMap map[string]*Connection
```

Comment:
If we are using the IP to index connections, define a key() function for type `Connection` that return c.ToIP. And is the FromIP used in any way?

Reply:
OK, sounds great. I have checked that node.Name is unique in the same namespace in k8s define. Using ip as key is good.

The FromIP does not used in current code and just used for marking, I will remove it later.